src/util.ts

```diff
@@ -66,3 +72,169 @@ export const getEncodedSHA256Hash = (str: string) => {
+      const startPosition = new vscode.Position(lineNumber, match.index);
+      const endPosition = new vscode.Position(
+        lineNumber,
+        match.index + match[0].length
+      );
+      comments.push(new vscode.Range(startPosition, endPosition));
+    }
+  }
+
+  return comments;
+}
+
+/**
+ * Convert SVG xml to png base64 url
+ * @param {any} svgXml
+ */
+export function getImageDataURL(svgXml: string) {
+  let base64 = encodeURIComponent(
+    Buffer.from(svgXml, "utf8").toString("base64")
+  );
+  return "data:image/svg+xml;base64," + base64;
+}
+
+export async function viewMermaidChart(
+  mcAPI: MermaidChartVSCode,
+  uuid: string
+) {
+  const panel = vscode.window.createWebviewPanel(
+    "mermaidChartView",
+    `Mermaid Chart: ${uuid}`,
+    vscode.ViewColumn.One,
+    {}
+  );
+
+  const isDarkTheme =
+    vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.Dark; //ColorTheme.Light;
+  // Choose the appropriate URL based on the current theme
+  const themeParameter = isDarkTheme ? "dark" : "light";
+  const svgContent = await mcAPI.getRawDocument(
+    {
+      documentID: uuid,
+      major: "0",
+      minor: "1",
+    },
+    themeParameter
+  );
+
+  panel.webview.html = `
+    <!DOCTYPE html>
+    <html lang="en" style="height: 100%;">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    </head>
+    <body style="height: 100%; margin: 0; padding: 0; overflow: hidden;">
+        <iframe sandbox="allow-same-origin allow-forms allow-popups allow-pointer-lock allow-top-navigation-by-user-activation" src="${getImageDataURL(
+          svgContent
+        )}" style="width: 100%; height: 100%; border: none;"></iframe>
+    </body>
+    </html>`;
+}
+export async function editMermaidChart(
+  mcAPI: MermaidChartVSCode,
+  uuid: string,
+  provider: MermaidChartProvider
+) {
+  const project = provider.getProjectOfDocument(uuid);
+  const projectUuid = project?.uuid;
+  if (!projectUuid) {
+    vscode.window.showErrorMessage(
+      "Diagram not found in project. Diagram might have moved to a different project."
+    );
+    return;
+  }
+  const editUrl = await mcAPI.getEditURL({
+    documentID: uuid,
+    projectID: projectUuid,
+    major: "0",
+    minor: "1",
+  });
+  vscode.env.openExternal(vscode.Uri.parse(editUrl));
+}
+
+export async function insertMermaidChartToken(
+  uuid: string,
+  provider: MermaidChartProvider
+) {
+  // If a project is selected from tree-view, no token shall be inserted
+  const itemType = provider.getItemTypeFromUuid(uuid);
+  if (itemType !== ITEM_TYPE_DOCUMENT) {
+    return;
+  }
+
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    return;
+  }
+  const mermaidChartTokenLine1 = `// [MermaidChart: ${uuid}]`;
+  editor.edit((editBuilder) => {
+    editBuilder.insert(
+      new vscode.Position(editor.selection.active.line, 0),
+      `${mermaidChartTokenLine1}\n`
+    );
```

Comment:
This makes sure we are adding the comment above a line and not in the middle of a line.

Reply:
OK